src/qibo/backends/abstract.py

```diff
@@ -15,6 +16,13 @@ def __init__(self):
         self.cpu_devices = []
         self.gpu_devices = []
         self.default_device = []
+        self.nthreads = None
+        if "OMP_NUM_THREADS" not in os.environ:
```

Comment:
I think would be great to include the `NUMBA_NUM_THREADS` environment flag, so we do not break the layout when e.g. running numba with tbb.

Reply:
Good point. I moved the `OMP_NUM_THREADS` in the qibotf backend as it is relevant only for this and added the check for `NUMBA_NUM_THREADS` in the numba backend. If None of these flags are given it will fall back to the default value from psutil.

Reply:
Perfect, thanks.